tests/rados/test_cephdf.py

```diff
@@ -286,3 +293,185 @@ def run(ceph_cluster, **kw):
+            "\nReef: BZ-2296248"
+            "\nSquid: BZ-2296247"
+            "\nThis test is to verify that ceph df MAX AVAIL is displayed correctly for all the pools"
+            " when OSD size is increased"
+            "Steps- \n"
+            "1. Creating a pool with default config\n"
+            "2. Log pool stats and verify max_avail \n"
+            "3. Create LVMs on backup nodes \n"
+            "4. Add backup hosts to the cluster and deploy OSDs on them \n"
+            "5. Log pool stats and verify max_avail \n"
+            "6. Expand OSD size on one backup host \n"
+            "7. Log pool stats and verify max_avail \n"
+            "8. Write data to a pool to fill the cluster \n"
+            "9. Log pool stats and verify max_avail \n"
+            "10. Expand OSD size on other backup host \n"
+            "11. Log pool stats and verify max_avail"
+        )
+
+        log.info(desc)
+        df_config = config.get("cephdf_max_avail_osd_expand")
+        pool_name = "test-osd-expand"
+        lvm_list = {"node12": [], "node13": []}
+
+        try:
+            # pass without execution for Squid
+            if not rhbuild.startswith("7"):
+                log.info("Test is currently valid only for RHCS 7.x")
+                return 0
+
+            # create default pool with given name
+            rados_obj.create_pool(pool_name=pool_name)
+
+            initial_pool_stat = rados_obj.get_cephdf_stats(pool_name=pool_name)
+            log.info(f"{pool_name} pool stat: {initial_pool_stat}")
+
+            # execute max_avail check across the cluster
+            if not rados_obj.verify_max_avail():
+                log.error("MAX_AVAIL deviates on the cluster more than expected")
+            log.info("MAX_AVAIL on the cluster are as per expectation")
+
+            for node in ["node12", "node13"]:
+                # create 10G LVMs on backup nodes
+                node_obj = get_node_by_id(ceph_cluster, node)
+                empty_devices = rados_obj.get_available_devices(
+                    node_name=node_obj.hostname, device_type="hdd"
+                )
+                if len(empty_devices) < 1:
+                    log.error(
+                        f"Need at least 1 spare disks available on host {node_obj.hostname}"
+                    )
+                    raise Exception(
+                        f"One spare disk not available on host {node_obj.hostname}"
+                    )
+
+                # for dev in empty_devices:
+                lvm_list[node].append(
+                    create_lvms(
+                        node=node_obj, count=1, size="10G", devices=empty_devices[0]
+                    )
+                )
+                log.info(
+                    f"List of LVMs created on {node_obj.hostname}: {lvm_list[node]}"
+                )
+
+            # add backup hosts to the cluster and deploy OSDs
+            service_obj.add_new_hosts(add_nodes=["node12", "node13"], deploy_osd=False)
+
```

Comment:
please check the count of active OSDs via `get_active_osd_list ` and check the new OSD addition.

Reply:
Added code to check if atleast 2 new OSDs have been added